components/com_fabrik/models/form.php

```diff
@@ -2333,14 +2335,18 @@ public function validate()
 			$post[$key] = $val;
 		}
 
-		if (!empty($this->errors))
+		// (Bauer) This would always be run because $this-errors contains an array with keys for each element - so $this->errors will never be an empty array. 
+		// Isn't that why the $ok flag was created to begin with???
+		// if (!empty($this->errors))
+		if(!$ok)
 		{
 			FabrikWorker::getPluginManager()->runPlugins('onError', $this);
-		}
-
-		FabrikHelperHTML::debug($this->errors, 'form:errors');
-		//echo "<pre>";print_r($this->errors);exit;
-		$this->setErrors($this->errors);
+			FabrikHelperHTML::debug($this->errors, 'form:errors');
+			//echo "<pre>";print_r($this->errors);exit;
+		
+			// (Bauer) Does this even need to be done if $ok is true? 
+			$this->setErrors($this->errors);
```

Comment:
i agree with the change to if ($ok)... but $this->setErrors($this->errors) should be outside that if statement